### PR TITLE
Add new flag set-read-only-root-filesystem to set readOnlyRootFilesys…

### PR DESCRIPTION
**Pull request by kristofferchr**
Solves: #1741
Support to set readOnlyRootFilesystem for EventListener container

# Changes
Add new flag set-read-only-root-filesystem.
Sets container securityContext readOnlyRootFilesystem to true when new flag is set to true

Prior to this it was not possible to set readOnlyRootFilesystem for EventListener container. Aligns security towards azure aks best practices and industry

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Added a new flag `el-read-only-root-filesystem` to the `tekton-triggers-controller` container. This flag, which is set to `true` by default, configures the `EventListener` container's `securityContext.readOnlyRootFilesystem` to `true`. This change aligns with Azure AKS best practices and enhances security.
```

### CODE REVIEW

**Comment by kristofferchr** [comment]
Unsure if this should have docs. Could not find any for flag `el-security-context`

**Review by khrm — COMMENTED** [review]
/ok-to-test

**Comment by kristofferchr** [comment]
@khrm and/or @savitaashture does any of you have time to spare to review this?

**Comment by khrm** [comment]
I would review this tomorrow. Sorry for the delay.

**Comment by khrm** [comment]
@savitaashture Let's review this.

**Comment by savitaashture** [comment]
Hi @kristofferchr :wave: 
Sorry for the delay 
Will take a look at it

**Comment by kristofferchr** [comment]
> Hi @kristofferchr :wave: 
> 
> Sorry for the delay 
> 
> Will take a look at it

Hi 😃 great!

File: pkg/reconciler/eventlistener/resources/container.go [hunk 1]
```diff
@@ -70,6 +70,9 @@ func MakeContainer(el *v1beta1.EventListener, configAcc reconcilersource.ConfigA
 	if !cfg.Defaults.IsDefaultRunAsGroupEmpty {
 		containerSecurityContext.RunAsGroup = ptr.Int64(cfg.Defaults.DefaultRunAsGroup)
 	}
+	if *c.SetReadOnlyRootFilesystem {
+		containerSecurityContext.ReadOnlyRootFilesystem = ptr.Bool(true)
+	}
```
Comment (savitaashture) [inline pkg/reconciler/eventlistener/resources/container.go:75]: can we move this under `SetSecurityContext` block itself

I mean if `SetSecurityContext` 

if *c.SetSecurityContext {
                if *c.SetReadOnlyRootFilesystem {
		containerSecurityContext.ReadOnlyRootFilesystem = ptr.Bool(true)
	}
		containerSecurityContext = corev1.SecurityContext{
			AllowPrivilegeEscalation: ptr.Bool(false),
			Capabilities: &corev1.Capabilities{
				Drop: []corev1.Capability{"ALL"},
			},
			RunAsNonRoot: ptr.Bool(cfg.Defaults.DefaultRunAsNonRoot),
			SeccompProfile: &corev1.SeccompProfile{
				Type: corev1.SeccompProfileTypeRuntimeDefault,
			},
		}
	}

Comment (kristofferchr) [reply pkg/reconciler/eventlistener/resources/container.go:75]: Fine by me. I will update the flag docs to note that the flag can only be used when el-security-context is set.

Comment (kristofferchr) [reply pkg/reconciler/eventlistener/resources/container.go]: Updated @savitaashture

**Review by savitaashture — APPROVED** [review]
Thank you @kristofferchr

**Comment by savitaashture** [comment]
> @savitaashture Let's review this.

@khrm can you do lgtm  again

**Review by khrm — COMMENTED** [review]
/lgtm

**Pull request merged**